crates/ruff_linter/src/rules/flake8_pie/rules/reimplemented_container_builtin.rs

```diff
@@ -31,46 +32,50 @@ use crate::checkers::ast::Checker;
 /// @dataclass
 /// class Foo:
 ///     bar: list[int] = field(default_factory=list)
+///     baz: dict[str, int] = field(default_factory=dict)
 /// ```
 ///
 /// ## References
 /// - [Python documentation: `list`](https://docs.python.org/3/library/functions.html#func-list)
 #[violation]
-pub struct ReimplementedListBuiltin;
+pub struct ReimplementedContainerBuiltin;
 
-impl Violation for ReimplementedListBuiltin {
+impl Violation for ReimplementedContainerBuiltin {
     const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
 
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Prefer `list` over useless lambda")
+        format!("Prefer `list` or `dict` over useless lambda")
     }
```

Comment:
You could store the type on the `Violation` then say "Prefer `list` over lambda" or "Prefer `dict` over lambda" depending on the case. It seems confusing to present both to the user when we know which they meant.

Reply:
Agreed! Can grep for `DeferralKeyword` as an example of this pattern.